src/main/java/com/checkmarx/jenkins/CxScanBuilder.java

```diff
@@ -2449,8 +2452,8 @@ public FormValidation doCheckScaSASTProjectID(@QueryParameter String value, @Que
          * @return
          */
         @POST
-        public FormValidation doCheckCustomFields(@QueryParameter String value) {
-            Jenkins.getInstance().checkPermission(Item.CONFIGURE);
+        public FormValidation doCheckCustomFields(@QueryParameter String value,@AncestorInPath Item item) {
+                item.checkPermission(Item.CONFIGURE);
```

Comment:
@Kevin-CB , are you saying for all job level parameters , we need to add the context check like we have added for all common parameters? But when we debugged the code for job level parameters the item is never coming as null it has value either a FreeStyleJobProject or PipelineJobProject.
![image](https://user-images.githubusercontent.com/75673851/166441411-e83fa01c-8f89-4ffe-af21-a47d9bbc0786.png)

Reply:
Hello @SubhadraSahoo, indeed to avoid any exception, we prefer that the case of a null context is always taken into account.
Because all web methods can be called from various Item.

Take for example this web method `doCheckCustomFields`, it can be called by its path: `descriptorByName/com.checkmarx.jenkins.CxScanBuilder/checkCustomFields`.

eg: `[JENKINS_INSTANCE]/descriptorByName/com.checkmarx.jenkins.CxScanBuilder/checkCustomFields` (context is null) or `[JENKINS_INSTANCE]/job/[JOB_NAME]descriptorByName/com.checkmarx.jenkins.CxScanBuilder/checkCustomFields` (the job is your context)

Reply:
@Kevin-CB Two questions.
1. Isn't it  an issue that Jenkins allowing access to doCheckCustomFields method via descriptor, when the function is only for the Job level field and there is no field by that name on the descriptor (global)?
2. Why an null pointer exception a concern for the security issue we are trying to fix?

Reply:
Hello @umeshwaghode,

1. No this is not a issue, it's used for many features, it's just necessary to implement the right permission checks when using them.

2. We're just trying to spread the best practices to prevent bad ones from spreading throughout the ecosystem. Many maintainers copy already existing code.

Reply:
"It is used in many features" does not explain how a different path should gives access to a function that is not meant for that path. I find that odd.

Reply:
That's how Jenkins (in conjunction with Stapler) is designed. Multiple URLs route to descriptor lists, as they may need to be available in multiple places. Even your build step's descriptor may offer global configuration options (through `global.jelly`) with related form validation, so not having the descriptor available at a global level would be a problem.

Reply:
Thanks Daniel for clarifying this further. We will be making suggested changes as without those I do not see we getting this PR approved.  But to clarify, all these places where (item == null) check is suggested in review comments, exist only on job and we have no intention to expose those in global.jelly.

Reply:
@Kevin-CB , Hi  Can you please remove the security warning from the Jenkins checkmarx plugin. The latest plugin 2022.2.3 is now published in the market place. Please let us know   if anything else  we need to do.